### PR TITLE
Add Linux appdata files

### PR DESCRIPTION
**Pull request by kirbyfan64**

### CODE REVIEW

File: app/build/resources/linux/mailspring.appdata.xml.in [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    </p>
+    <p>
+      Mailspring comes packed with powerful features like Unified Inbox, Snooze, Send Later, Mail
+      Rules, Templates and more. Mailspring Pro, which you can unlock with a monthly subscription,
+      adds even more features for people who send a ton of email: link tracking, read receipts,
+      mailbox analytics, contact and company profiles. All of these features run in the client -
+      Mailspring does not send your email credentials to the cloud.
+    </p>
+  </description>
+  <categories>
```
Comment (bilelmoussaoui) [inline app/build/resources/linux/mailspring.appdata.xml.in]: Should be put in the desktop file instead.

**Comment by kirbyfan64** [comment]
@bilelmoussaoui Done.

File: app/build/resources/linux/mailspring.desktop.in [hunk 2]
```diff
@@ -7,5 +7,5 @@ Icon=mailspring
 Type=Application
 StartupNotify=true
 StartupWMClass=<%= productName %>
-Categories=GNOME;GTK;Network;Email;
+Categories=GNOME;GTK;Network;Email;Productivity;Communication;
```
Comment (bilelmoussaoui) [inline app/build/resources/linux/mailspring.desktop.in:10]: How does even mailspring assign itself as a GNOME & Gtk application? :o

Comment (refi64) [reply app/build/resources/linux/mailspring.desktop.in:10]: ¯\\\_(ツ)\_/¯ I'm just trying to do the appdata file 😅

Comment (bilelmoussaoui) [reply app/build/resources/linux/mailspring.desktop.in:10]: Never said it was a comment for you 😅

Comment (bengotow) [reply app/build/resources/linux/mailspring.desktop.in:10]: 👋 hey guys! You know honestly, I think I copied this file from somewhere to create it ages and ages ago - I'm really not sure what those GNOME + GTK entries do, if anything. Thanks for fixing this @bilelmoussaoui!

**Comment by bengotow** [comment]
Hey! This looks great, thanks for adding the appdata file. Does it need to be copied to a specific path during the installation process on Linux? I think in the `installer-linux-task.js` file we write `mailspring.desktop.in` to an "output" folder, and then there are two scripts that pack the output folder into .deb and .rpm packages at `app/script/mkdeb` and `app/script/mkrpm`. 

It looks like the `mkdeb` script has these line in it to move the mailspring.desktop file to a path inside the debian package root:

```
mkdir -m $FILE_MODE -p "$TARGET/usr/share/applications"
cp "$OUTPUT_PATH/mailspring.desktop" "$TARGET/usr/share/applications"
```

I wonder if the mkdeb script needs to move the appdata file somewhere else as well?

(Also sidenote, I know this is sort of weird, but the snapcraft ".snap" file is built from the .deb package, so there isn't a `mksnap` script.)

File: app/build/resources/linux/mailspring.desktop.in (hunk 2 above)
Comment (bilelmoussaoui) [reply app/build/resources/linux/mailspring.desktop.in:10]: @bengotow those tags are used either on a search of the system itself or in the software center. The application wasn't made using GNOME technologies (Gtk and other libraries around it), those two categories could be removed from there

**Comment by kirbyfan64** [comment]
@bengotow Ah, whoops. Fixed.

**Comment by bengotow** [comment]
Looks great! Thanks for addressing the comments. There will be a patch release in a few days so this will get shipped soon!

**Pull request merged**